src/test/java/emissary/test/core/junit5/ExtractionTest.java

```diff
@@ -391,6 +411,25 @@ protected void checkStringValue(Element meta, String data, String tname) {
         }
     }
 
+    protected boolean verifyOs(Element element) {
+        Attribute specifiedOs = element.getAttribute("os-release");
+        if (specifiedOs != null) {
+            String os = specifiedOs.getValue();
+            switch (os) {
+                case "ubuntu":
+                    return OSReleaseUtil.isUbuntu();
```

Comment:
I have concerns with reading a file each time an OS check is performed. Consider storing the OS release as a static field.

Reply:
added in a static field to store the found OS on the first check, then compare the attribute tags to that field on other checks.